### PR TITLE
User with no privileges for repository can view and modify assets in that repository

### PR DESCRIPTION
**Pull request by ederign**

### CODE REVIEW

File: kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-api/src/main/java/org/kie/workbench/common/screens/social/hp/service/RepositoryListService.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package org.kie.workbench.common.screens.social.hp.service;
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-api/src/main/java/org/kie/workbench/common/screens/social/hp/service/RepositoryListService.java]: License!

File: kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-backend/src/main/test/org/kie/workbench/common/screens/social/hp/service/RepositoryListServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
+package org.kie.workbench.common.screens.social.hp.service;
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-backend/src/main/test/org/kie/workbench/common/screens/social/hp/service/RepositoryListServiceImplTest.java]: License!

File: kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-backend/src/main/java/org/kie/workbench/common/screens/social/hp/service/RepositoryListServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
+package org.kie.workbench.common.screens.social.hp.service;
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-backend/src/main/java/org/kie/workbench/common/screens/social/hp/service/RepositoryListServiceImpl.java]: License!

File: kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-backend/src/main/test/org/kie/workbench/common/screens/social/hp/service/RepositoryListServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+    @Test
+    public void getRepositories() {
+        final List<String> repositories = service.getRepositories();
+        assertEquals( 1, repositories.size() );
+        assertEquals( "dora", repositories.get( 0 ) );
+
+    }
+
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-backend/src/main/test/org/kie/workbench/common/screens/social/hp/service/RepositoryListServiceImplTest.java]: How about a test filtering unauthorised Repositories too?

**Comment by manstis** [comment]
@ederign Other than a few minor comments, this gets a +1 from me.

**Comment by ederign** [comment]
@manstis licenses added! The unauthorised  repositories tests are on socialEventRepositoryConstraint.

@csadilek @porcelli please wait for merge because I'll probably will need this in 6.3

**Comment by manstis** [comment]
@ederign ... well... you test "authorised" Repositories with your mock ```SocialEventRepositoryConstraint``` but you don't test for "unauthorised"... e.g. have another test where ```SocialEventRepositoryConstraint.getAuthorizedRepositories()``` returns an empty collection. Just a detail.. I know and you know it should work.. but isn't that the point of tests? ;)

**Comment by manstis** [comment]
+1 to merge (following discussion with @ederign on IRC)

**Comment by ederign** [comment]
6.3: https://github.com/droolsjbpm/kie-wb-common/pull/227

**Comment by ederign** [comment]
6.4: https://github.com/droolsjbpm/kie-wb-common/pull/228

**Comment by csadilek** [comment]
master: ac67f4f1bb4a676a552e1d97670d227ac68b8ab4

**Pull request closed without merging**